src/v/storage/offset_to_filepos_consumer.h

```diff
@@ -60,6 +60,11 @@ class offset_to_filepos_consumer {
                 _target_last_offset,
                 _accumulator)});
         }
+        /**
+         * We update _filepos here in case we hit a gap, if so we need to return
+         * an offset and end position of a last batch preceding the gap.
+         */
+        _filepos = {batch.last_offset(), _accumulator};
```

Comment:
I don't understand how we can successfully return this. Say, we have two batches, [0-10] and [15-20] and target=12. I think we will process 0-10 and then blow up on the check in line 52.

Reply:
The answer was in `reader_config` - i.e. if the target falls into the gap, we'll stop reading after we reach the gap and return the `_filepos`